src/clj/puppetlabs/puppetserver/certificate_authority.clj

```diff
@@ -570,29 +632,37 @@
     [infra-file :- schema/Str]
     (line-seq (io/reader infra-file)))
 
+(defn- write-infra-serials-to-writer
+  [writer infra-nodes-path infra-node-serials-path signeddir]
+  (try
+    (let [infra-nodes (read-infra-nodes infra-nodes-path)]
+      (doseq [infra-node infra-nodes]
+        (try
+          (let [infra-serial (-> (path-to-cert signeddir infra-node)
+                                 (utils/pem->cert)
+                                 (utils/get-serial))]
+            (.write writer (str infra-serial))
+            (.newLine writer))
+          (catch java.io.FileNotFoundException ex
+            (log/warn
+             (i18n/trs
+              (str
```

Comment:
I don't think you need to wrap this in `str` - `trs` should let you interpolate with `{0}`.

Reply:
That's true, but it's fine either way and this isn't code that actually changed in this PR so I'm not going to touch it.

Reply:
fair